src/AI/AIUtils.java

```diff
@@ -280,13 +280,14 @@ public static double scoreAttackAction(Unit unit, GameAction action, GameMap map
     ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(unit, gameMap, includeOccupiedDestinations);
     for( WeaponModel wep : unit.model.weapons )
     {
-      if( wep.getDamage(target) > 0 )
+      double damage = (null == target)? 1 : wep.getDamage(target);
```

Comment:
Why move `wep.getDamage(target)` here and not `unit.getHPFactor()` also? The resulting damage term is only used with `getHPFactor()` to calculate `val`.

Also why multiply `damage` with hp factor in the first place? A weapon that can do 5 damage to a 10HP unit is even more dangerous to a 6HP unit, but this would make it have a lower `val` in the second case.

Reply:
> Why move `wep.getDamage(target)` here and not `unit.getHPFactor()` also? The resulting damage term is only used with `getHPFactor()` to calculate `val`.

Fair. Mostly, I just wanted to enable using it with a `null` target to get all weapons.
> Also why multiply `damage` with hp factor in the first place? A weapon that can do 5 damage to a 10HP unit is even more dangerous to a 6HP unit, but this would make it have a lower `val` in the second case.

It's the *attacker's* HP factor. A 1 HP tank always has the same result from `WeaponModel.getDamage()`, but deals substantially less *real* damage because of HP.

Reply:
> attacker's HP factor

Oh. Right.